libbeat/dashboards/export.go

```diff
@@ -57,12 +57,12 @@ func Export(client *kibana.Client, id string) (common.MapStr, error) {
 func ExportAllFromYml(client *kibana.Client, ymlPath string) ([]common.MapStr, ListYML, error) {
 	b, err := ioutil.ReadFile(ymlPath)
 	if err != nil {
-		return nil, ListYML{}, fmt.Errorf("error opening the list of dashboards: %+v", err)
+		return nil, ListYML{}, errors.Wrap(err, "error opening the list of dashboards")
```

Comment:
AFAIK we are trying to stick with using `fmt.Errorf` and `%+v` to report errors.

Reply:
By using `errors.Wrap` you retain the contextual information about the original error and you add additional context added to the returned error about the current stack. If you later print the error with `%+v` you will get the extended format that contains details about each frame in the stacktrace. But the caller get to choose whether or not it wants to log a message that contains the extended details with a stack trace.

Reply:
Got it partially. How is then an enhancement to the current error handling? AFAIK it prints the same thing when logged. Is that correct?

Reply:
There will be a difference if the error is logged with `%+v` (which it is) because now that it's wrapped it contains a call stack information and the `error` object implements a `Format` to know when to react to `%+v` vs `%v`.

Before (but only with modifications because before the error was unchecked):
```
$ go run ../dev-tools/cmd/dashboards/export_dashboards.go -yml doesntexist.yml
Failed to export dashboards from YML file: error opening the list of dashboards: open doesntexist.yml: no such file or directory
exit status 1
```

After
```
$ go run ../dev-tools/cmd/dashboards/export_dashboards.go -yml doesntexist.yml
Failed to export dashboards from YML file: open doesntexist.yml: no such file or directory
error opening the list of dashboards
github.com/elastic/beats/libbeat/dashboards.ExportAllFromYml
	/Users/akroh/go/src/github.com/elastic/beats/libbeat/dashboards/export.go:61
main.exportDashboardsFromYML
	/Users/akroh/go/src/github.com/elastic/beats/dev-tools/cmd/dashboards/export_dashboards.go:95
main.main
	/Users/akroh/go/src/github.com/elastic/beats/dev-tools/cmd/dashboards/export_dashboards.go:76
runtime.main
	/Users/akroh/.gvm/versions/go1.11.4.darwin.amd64/src/runtime/proc.go:201
runtime.goexit
	/Users/akroh/.gvm/versions/go1.11.4.darwin.amd64/src/runtime/asm_amd64.s:1333
exit status 1
```

Reply:
I prefer that that too, but the logger statements were already using `%+v` so I figured that it was the author's intention to log them. So I guess if that was you then it wasn't intentional. I can change the `log.Fatalf` statements if that's good with you?

Reply:
Perfect. Thanks.